systemtest/src/main/java/io/strimzi/systemtest/utils/UpgradeDowngradeDatalist.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.utils;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.type.CollectionType;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import io.strimzi.test.TestUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class UpgradeDowngradeDatalist {
+    private static final Logger LOGGER = LogManager.getLogger(UpgradeDowngradeDatalist.class);
+
+    private List<UpgradeDowngradeData> data;
+
+    public UpgradeDowngradeDatalist(String fileName) {
+        try {
+            ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+            CollectionType modificationDataListType = mapper.getTypeFactory().constructCollectionType(List.class, UpgradeDowngradeData.class);
+            List<UpgradeDowngradeData> datalist = mapper.readValue(new File(TestUtils.USER_PATH + "/src/test/resources/upgrade/" + fileName), modificationDataListType);
+
+            if (fileName.contains("Upgrade")) {
+                datalist.forEach(upgradeData -> {
+                    // Set upgrade destination data to latest version which is HEAD
+                    upgradeData.setUrlTo("HEAD");
+                    upgradeData.setToVersion("HEAD");
+                    upgradeData.setToExamples("HEAD");
+                });
+            }
```

Comment:
Wouldn't be better just load both files automatically to different Lists and then just access upgrade/downgrade data?